### PR TITLE
feat: Add Toast feature

### LINKED ISSUE #2664: Feedback when copying to clipboard
Would be great to add some kind of toast when we do shift-option-c to copy to clipboard. It's not clear that it's working right now.

### PR DESCRIPTION
**Pull request by rish07**
Closes #2664 

@vjeux @lipis I have tried something and have been able to accomplish the goal. Please let me know any changes that you want me to make. 

Thank you!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,8 +32,10 @@
     "browser-nativefs": "0.12.0",
     "clsx": "1.1.1",
     "firebase": "8.2.1",
+    "framer-motion": "3.1.1",
```
Comment (lipis) [inline package.json:35]: https://bundlephobia.com/result?p=framer-motion@3.1.1

File: package.json [hunk 2]
```diff
@@ -44,6 +47,7 @@
     "pwacompat": "2.0.17",
     "react": "17.0.1",
     "react-dom": "17.0.1",
+    "react-feather": "2.0.9",
```
Comment (lipis) [inline package.json:50]: https://bundlephobia.com/result?p=react-feather@2.0.9

**Comment by lipis** [comment]
First of all thank you for taking up the challenge!

A few things:
- We are very careful when adding new libraries, and this feature doesn't seem to need any
- We should keep it simpler as @vjeux mentioned and not use any external library
- Move the component to the [`components`](https://github.com/excalidraw/excalidraw/tree/master/src/components) directory
- For the final look and feel, we'll iterate later.

File: src/components/App.tsx [hunk 3]
```diff
@@ -270,6 +274,111 @@ export type ExcalidrawImperativeAPI = {
...
+      delete data.toasts[id];
+    },
+  },
+});
+
+interface MessageProps {
+  toast: ToastMessage;
+}
+
+function Toast({ toast }: MessageProps) {
```
Comment (lipis) [inline src/components/App.tsx:312]: use arrow functions in general
```suggestion
const Toast = ({ toast }: MessageProps) => {
```

**Comment by rish07** [comment]
> First of all thank you for taking up the challenge!
> 
> A few things:
> 
> * We are very careful when adding new libraries, and this feature doesn't seem to need any
> * We should keep it simpler as @vjeux mentioned and not use any external library
> * Move the component to the [`components`](https://github.com/excalidraw/excalidraw/tree/master/src/components) directory
> * For the final look and feel, we'll iterate later
> * All the copy should be added to the [`en.json`](https://github.com/excalidraw/excalidraw/blob/master/src/locales/en.json) and use it like: `t("labels.copy")` for example (to be able to support other languages). Maybe a new section for these hints?

Thanks for the review @lipis. I am having problems in implementing toasts from scratch in typescript. Are there any code samples or resources you could link here so I can take a reference from there?

**Comment by dwelle** [comment]
Thanks @rish07 for the PR!

I'm with @lipis — I don't think we should pull in half a megabyte of libs (minified) just for this one feature. CSS should be enough for this. If we stick to current implem where we don't display more than 1 toast at a time, it should be even simpler. I'm sure we'll figure it out.

Sidenotes:

- I'd not show toasts for copy/cut actions as that's pretty obvious.
- Do we want to show toasts when the action isn't invoked via a shortcut? E.g. when I click on `copy as svg` via contextmenu, I don't think I need a toast confirmation for this.

**Comment by lipis** [comment]
We have a simpler toast in another PR.. so we'll close this one. Thanks for the effort. #2772

**Pull request closed without merging**